catalog/core/catalog-core-encryption/src/main/java/org/codice/ddf/catalog/content/encryption/ContentEncryptorDecryptor.java

```diff
@@ -0,0 +1,275 @@
+  private static final String CONTENT_ENCRYPTION_KEY_ALIAS_VALUE =
+      SecurityConstants.getContentEncryptionKeyAlias();
+  private static final String CONTENT_ENCRYPTION_KEY_PASSWORD_VALUE =
+      SecurityConstants.getContentEncryptionKeyPassword();
+  private static final String AES = "AES";
+  private static final String CIPHER_TYPE = "AES/GCM/NoPadding";
+  private static final String ENABLE_DEBUG_LOGGING_MSG =
+      " Enable debug logging for more information.";
+  private static final int KEY_LENGTH = 256;
+  @VisibleForTesting static final int IV_BYTE_SIZE = 12;
+  @VisibleForTesting static final int AUTH_TAG_LENGTH = 128;
+
+  private static SecureRandom secureRandom = new SecureRandom();
+  private static Cipher cipher;
+  private static KeyStore keyStore;
+  private static KeyStore trustStore;
+  private static SecretKey contentEncryptionKey;
+
+  public ContentEncryptorDecryptor() {
+    initKeystores();
+    initKey();
+    initCipher();
+  }
+
+  @VisibleForTesting
+  ContentEncryptorDecryptor(SecretKey secretKey) {
+    contentEncryptionKey = secretKey;
+    initCipher();
+  }
+
+  @Override
+  public byte[] encrypt(InputStream inputStream) throws ContentEncryptionException {
+    try {
+      return encrypt(ByteStreams.toByteArray(inputStream));
+    } catch (IOException e) {
+      LOGGER.warn("Problem reading input stream.");
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] encrypt(byte[] inputBytes) throws ContentEncryptionException {
+    try {
+      readyCipherForEncryption();
+      return wrapWithIV(cipher.getIV(), cipher.doFinal(inputBytes));
+    } catch (BadPaddingException | IllegalBlockSizeException e) {
+      LOGGER.warn("Problem encrypting file content." + ENABLE_DEBUG_LOGGING_MSG);
+      LOGGER.debug("", e);
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] decrypt(InputStream inputStream) throws ContentEncryptionException {
+    try {
+      return decrypt(ByteStreams.toByteArray(inputStream));
+    } catch (IOException e) {
+      LOGGER.warn("Problem reading input stream.");
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  @Override
+  public byte[] decrypt(byte[] inputBytes) throws ContentEncryptionException {
+    try {
+      // separate the iv from the encrypted payload
+      ByteBuffer byteBuffer = ByteBuffer.wrap(inputBytes);
+      byte[] iv = new byte[IV_BYTE_SIZE];
+      byte[] encryptedBytes = new byte[inputBytes.length - IV_BYTE_SIZE];
+      byteBuffer.get(iv);
+      byteBuffer.get(encryptedBytes);
+
+      readyCipherForDecryption(iv);
+      return cipher.doFinal(encryptedBytes);
+    } catch (BadPaddingException | IllegalBlockSizeException e) {
+      LOGGER.warn("Problem decrypting file content." + ENABLE_DEBUG_LOGGING_MSG);
+      LOGGER.debug("", e);
+      throw new ContentEncryptionException(e);
+    }
+  }
+
+  private static void initKeystores() {
```

Comment:
After trying to implement this, I think keeping the method the way it is now is a better solution. I would have to make a good amount of design changes in order to let this method do what I want it to do. E.g.:
```
      keyStore = SecurityConstants.newKeystore();
      trustStore = SecurityConstants.newTruststore();
```
would be very difficult to call conditionally off of an argument in the new method. I _could_ pass in what security constant I want to use, but that would require me passing 3+ arguments to the new method for all of the different constants needed.

Reply:
I was thinking something as simple as this:
```java
  private static void initKeystores() {
    try {
      keyStore = SecurityConstants.newKeystore();
      trustStore = SecurityConstants.newTruststore();
    } catch (KeyStoreException e) {
      LOGGER.info(
          "Unable to create keystore instance of type {}",
          System.getProperty(SecurityConstants.KEYSTORE_TYPE),
          e);
    }

    if (initKeystore(
        keyStore,
        Paths.get(SecurityConstants.getKeystorePath()),
        SecurityConstants.getKeystorePassword().toCharArray())) {
      initKeystore(
          trustStore,
          Paths.get(SecurityConstants.getTruststorePath()),
          SecurityConstants.getTruststorePassword().toCharArray());
    }
  }

  private static boolean initKeystore(KeyStore keyStore, Path keyStorePath, char[] password) {
    Path ddfHomePath = Paths.get(System.getProperty("ddf.home"));
    if (!keyStorePath.isAbsolute()) {
      keyStorePath = Paths.get(ddfHomePath.toString(), keyStorePath.toString());
    }
    if (!Files.isReadable(keyStorePath)) {
      LOGGER.warn("Unable to read system key store file, {}", keyStorePath);
      return false;
    }

    try (InputStream kfis = Files.newInputStream(keyStorePath)) {
      keyStore.load(kfis, password);
    } catch (NoSuchAlgorithmException | CertificateException | IOException e) {
      LOGGER.warn(
          "Unable to load system keystore file, {} {}", keyStorePath, ENABLE_DEBUG_LOGGING_MSG);
      LOGGER.debug("Unable to load system keystore file", e);
      try {
        keyStore.load(null, null);
      } catch (NoSuchAlgorithmException | CertificateException | IOException ignore) {
      }
    }

    return true;
  }
```
The only thing you lose with that is some distinguishing `keyStore` vs `trustStore` on logs which has limited value.

Reply:
Why do you even need to get the truststore for anything? We're doing AES and that is symmetric. Also, any code that accesses the keystores should be wrapped in a doPrivileged block so that the access isn't bled up the call stack for permissions.

Reply:
Since we took out the truststore I kept the old logic and modified it to only initialize the keystore.